### DISCUSSION TITLE
Get all orders on resolution

### DISCUSSION THREAD

**Discussion by Wulfheart** [zond/godip#125] [General]
If I don't provide any orders godip assumes some orders. Is it possible to get all orders including the assumed orders after running ``state.Next()``?

**Comment by zond** [comment]
Not currently, no. It would be kind of neat if the game server showed those orders as well... Maybe we want that in Diplicity too.

Currently, you'd have to do what the State does at https://github.com/zond/godip/blob/master/state/state.go#L191 and https://github.com/zond/godip/blob/master/state/state.go#L207 to get those orders, but I agree there should be a better way.

@tttppp, please chime in if you have opinions here!

Some alternatives I see:

1. The Phase at https://github.com/zond/godip/blob/master/godip.go#L314 gets a function that returns orders of the phase, that is allowed to return an empty set pre-adjudication.
2. Add something to State, or State.Next, that gives us the orders. But what should that look like?

Opinions?

**Reply by Wulfheart** [reply]
I did a pull request on that. #126 
It just adds a property ``previouslyAppliedOrders`` which is accessible via a getter method. In ``state.Next()`` it copies before the execution of the movements all orders into``previouslyAppliedOrders``.
